runtime/gc_glue_java/ArrayletObjectModel.hpp

```diff
@@ -843,13 +843,16 @@ class GC_ArrayletObjectModel : public GC_ArrayletObjectModelBase
 	MMINLINE void **
 	dataAddrSlotForContiguous(J9IndexableObject *arrayPtr)
 	{
-		AssertContiguousArrayletLayout(arrayPtr);
-		bool const compressed = compressObjectReferences();
 		void **dataAddrPtr = NULL;
-		if (compressed) {
-			dataAddrPtr = &((J9IndexableObjectContiguousCompressed *)arrayPtr)->dataAddr;
-		} else {
-			dataAddrPtr = &((J9IndexableObjectContiguousFull *)arrayPtr)->dataAddr;
+
+		if (static_cast<J9JavaVM*>(_omrVM->_language_vm)->isDataAddrPresent) {
```

Comment:
Is this really necessary? This should not be even called by a configuration that does not have dataAddr present. You can try converting it into an assert....

Same for dataAddrSlotForDiscontiguous.

Reply:
Yes, that's correct it is not really necessary. I added it to prevent a crash if someone else was to add a call to this function without understanding that the dataAddr field might not be there, but I can remove it.

Reply:
Yes, I'd completely remove it. If somehow this is mistakenly used, we will crash soon (hence we can easily catch the problem in regular testing)